### DISCUSSION TITLE
VSCode browser preview resets URL after reload

### DISCUSSION THREAD

**Discussion by m-guggenmos** [quarto-dev/quarto-cli#6330] [Q&A]
### Description

I am playing around with revealjs and hit the following problem: when using the internal VSCode live preview, the URL is correctly updated with the anchor (`#xyz`) for the current slide. However, when I make a change to a qmd file, the built-in browser reloads to the initial value of the URL and I have to click back to the slide. The same happens, when I click the reload button of the browser. This does only happen in the built-in VSCode browser and not in external browsers. This makes working with the build-in browser very uncomfortable.

Any ideas or what would be the best place to look for help with this issue?

This is on a Ubuntu Linux system.

**Comment by mcanouil** [comment]
Would you be able to share some steps to reproduce the issue you are describing?

**Reply by m-guggenmos** [reply]
Sure.

`_quarto.yml`
```
project:
  output-dir: _slides
  render:
    - test.qmd

title: "Title"
author: "John Doe"

format: revealjs
```

`test.qmd`
```
## Heading

Text

## Heading2

Text

## Heading3

Text

## Heading4

Text
```

Quarto settings:
Quarto > Render: Preview Type: internal

1. Start preview, e.g. by clicking "Render" in VSCode on the `test.qmd` file. 
2. Navigate to Heading4 (-> the URL correctly updates to `test.html#/heading4`)
3. Make a change to the text in `test.qmd` under `## Heading4`. Save file.
4. The website preview refreshes. However, for me, it jumps back to the title page (the URL also changes to the title page's URL/anchor) and I can't live monitor the changes I have made to the current slide with Heading4. I remember that sometimes it also jumped back to the previous slide, but I can't reproduce this atm.

**Reply by mcanouil** [reply]
I don't use "render on save" as you seems to do.

Anyway, I suggest you open an issue on the Quarto VSCode extension repository, see <https://github.com/quarto-dev/quarto/issues>.

**Reply by m-guggenmos** [reply]
This issue is independent of "render on save" and is also present when just manually reloading the Quarto Preview browser. I'll open an issue.

**Reply by m-guggenmos** [reply]
https://github.com/quarto-dev/quarto/issues/225
